lib/vsc/utils/rest.py

```diff
@@ -124,15 +125,18 @@ def head(self, url, headers=None, **params):
         url += self.urlencode(params)
         return self.request(self.HEAD, url, None, headers)
 
-    def delete(self, url, headers=None, **params):
+    def delete(self, url, headers=None, body=None, **params):
         """
-        Do a http delete request on the given url with given headers and parameters
+        Do a http delete request on the given url with given headers, body and parameters
         Parameters is a dictionary that will will be urlencoded
         """
         if self.append_slash:
             url += '/'
         url += self.urlencode(params)
-        return self.request(self.DELETE, url, None, headers)
+        if headers is None:
```

Comment:
please stop copy pasting the same data over and over. use  a small function or even a decrator. (or move the same code to the `self.request`)